firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/PersonalizationTest.java

```diff
@@ -88,43 +123,46 @@ public void logArmActive_nonPersonalizationKey_notLogged() {
 
     verify(mockAnalyticsConnector, times(0))
         .logEvent(
-            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
+            eq(ANALYTICS_ORIGIN_PERSONALIZATION),
+            or(eq(EXTERNAL_EVENT), eq(INTERNAL_EVENT)),
+            any(Bundle.class));
     assertThat(FAKE_LOGS).isEmpty();
   }
 
   @Test
-  public void logArmActive_singlePersonalizationKey_loggedOnce() {
+  public void logArmActive_singlePersonalizationKey_loggedInternallyAndExternally() {
     personalization.logArmActive("key1", CONFIG_CONTAINER);
 
     verify(mockAnalyticsConnector, times(1))
-        .logEvent(
-            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
-    assertThat(FAKE_LOGS).hasSize(1);
+        .logEvent(eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(EXTERNAL_EVENT), any(Bundle.class));
+    verify(mockAnalyticsConnector, times(1))
+        .logEvent(eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(INTERNAL_EVENT), any(Bundle.class));
+    assertThat(FAKE_LOGS).hasSize(2);
 
-    Bundle params = new Bundle();
-    params.putString(ARM_KEY, "id1");
-    params.putString(ARM_VALUE, "value1");
-    assertThat(FAKE_LOGS.get(0).toString()).isEqualTo(params.toString());
+    assertThat(FAKE_LOGS)
+        .comparingElementsUsing(TO_STRING)
+        .containsExactly(LOG_PARAMS_1.toString(), INTERNAL_LOG_PARAMS_1.toString());
   }
 
   @Test
-  public void logArmActive_multiplePersonalizationKeys_loggedMultiple() {
+  public void logArmActive_multiplePersonalizationKeys_loggedInternallyAndExternally() {
     personalization.logArmActive("key1", CONFIG_CONTAINER);
     personalization.logArmActive("key2", CONFIG_CONTAINER);
+    personalization.logArmActive("key1", CONFIG_CONTAINER);
 
     verify(mockAnalyticsConnector, times(2))
-        .logEvent(
-            eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(ANALYTICS_PULL_EVENT), any(Bundle.class));
-    assertThat(FAKE_LOGS).hasSize(2);
-
-    Bundle params1 = new Bundle();
-    params1.putString(ARM_KEY, "id1");
-    params1.putString(ARM_VALUE, "value1");
-    assertThat(FAKE_LOGS.get(0).toString()).isEqualTo(params1.toString());
-
-    Bundle params2 = new Bundle();
-    params2.putString(ARM_KEY, "id2");
-    params2.putString(ARM_VALUE, "value2");
-    assertThat(FAKE_LOGS.get(1).toString()).isEqualTo(params2.toString());
+        .logEvent(eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(EXTERNAL_EVENT), any(Bundle.class));
+    verify(mockAnalyticsConnector, times(2))
+        .logEvent(eq(ANALYTICS_ORIGIN_PERSONALIZATION), eq(INTERNAL_EVENT), any(Bundle.class));
+    assertThat(FAKE_LOGS).hasSize(4);
+
+    assertThat(FAKE_LOGS)
+        .comparingElementsUsing(TO_STRING)
+        .containsAtLeast(LOG_PARAMS_1.toString(), LOG_PARAMS_2.toString())
+        .inOrder();
+    assertThat(FAKE_LOGS)
+        .comparingElementsUsing(TO_STRING)
+        .containsAtLeast(INTERNAL_LOG_PARAMS_1.toString(), INTERNAL_LOG_PARAMS_2.toString())
```

Comment:
Seems like this could be combined with the above into one statement with all four logs?

Reply:
We can't combine it if we want to check the ordering too, since it's not guaranteed that the external ones come before the internal ones.